garak/resources/tap/tap_main.py

```diff
@@ -17,28 +18,29 @@
     extract_json,
 )
 from .system_prompts import attacker_system_prompt, on_topic_prompt, judge_system_prompt
-from .generator_utils import load_generator, supported_openai, supported_huggingface
+from .generator_utils import load_generator, supported_openai, supported_huggingface, token_count, get_token_limit
 
 import garak.generators
 import garak._config
+from garak.generators.huggingface import Model
 
 logger = getLogger(__name__)
 SAVE_RESULTS = True
 
 
 class AttackManager:
     def __init__(
-        self,
-        goal: str,
-        attack_generator: garak.generators.Generator,
-        target_generator: garak.generators.Generator,
-        evaluation_generator: garak.generators.Generator,
-        attack_max_tokens: int = 500,
-        attack_max_attempts: int = 5,
-        max_parallel_streams: int = 5,
-        target_max_tokens: int = 150,
-        evaluator_max_tokens: int = 10,
-        evaluator_temperature: float = 0.0,
+            self,
```

Comment:
hey do you know how these params can be set when invoking via the probe?

Reply:
I'd think you could pass them as params in the probe config?

Reply:
Looks like we need a `ProbeConfig` class in `_config`.

Reply:
values in `_config.plugins.probes.ClassName` are automatically set in classes at instantiation time - just wondering a bit about the stuff in `resources/`. maybe let's wait for demand before triaging dealing with that